hawc/services/epa/hero.py

```diff
@@ -90,24 +124,47 @@ def get_content(self):
             self.failures = []
             return dict(success=self.content, failure=self.failures)
 
+        if settings.HAWC_FEATURES.ENABLE_NEW_HERO:
+            if settings.HERO_API_KEY is None:
+                raise ValueError("HERO_API_KEY required.")
+            headers = {"Authorization": f"Bearer {settings.HERO_API_KEY}"}
+
+        parse_func = parse_article_new if settings.HAWC_FEATURES.ENABLE_NEW_HERO else parse_article
         rng = list(range(0, self.ids_count, self.settings["recordsperpage"]))
         for recstart in rng:
             request_ids = self.ids[recstart : recstart + self.settings["recordsperpage"]]
             ids = ",".join([str(id_) for id_ in request_ids])
             rpp = self.settings["recordsperpage"]
-            url = f"https://hero.epa.gov/hero/ws/index.cfm/api/1.0/search/criteria/{ids}/recordsperpage/{rpp}.json"
-            try:
-                r = requests.get(url, timeout=30.0)
-                if r.status_code == 200:
-                    data = json.loads(r.text)
-                    for ref in data["results"]:
-                        self.content.append(parse_article(ref))
-                else:
+            results = []
+            if settings.HAWC_FEATURES.ENABLE_NEW_HERO:
+                url = "https://heronetnext.epa.gov/api/reference/export/json"
+                data = {"type": "hero", "id": request_ids}
+                try:
+                    r = requests.post(url, json=data, headers=headers, timeout=30.0)
+                    if r.status_code == 200:
+                        results = r.json()
+                    else:
+                        logger.info(f"HERO request failure: {url}")
+                except requests.exceptions.Timeout:
+                    logger.info(f"HERO request timeout: {url}")
+                except json.JSONDecodeError:
                     logger.info(f"HERO request failure: {url}")
-            except requests.exceptions.Timeout:
-                logger.info(f"HERO request timeout: {url}")
-            except json.JSONDecodeError:
-                logger.info(f"HERO request failure: {url}")
+            else:
+                url = f"https://hero.epa.gov/hero/ws/index.cfm/api/1.0/search/criteria/{ids}/recordsperpage/{rpp}.json"
+                try:
+                    r = requests.get(url, timeout=30.0)
+                    if r.status_code == 200:
+                        data = json.loads(r.text)
+                        results = data["results"]
+                    else:
+                        logger.info(f"HERO request failure: {url}")
+                except requests.exceptions.Timeout:
+                    logger.info(f"HERO request timeout: {url}")
+                except json.JSONDecodeError:
+                    logger.info(f"HERO request failure: {url}")
+
+            for ref in results:
+                self.content.append(parse_func(ref))
```

Comment:
instead of adding an if statement to determine which parser to use, we select a `parse_func` and then use it everywhere. I just made a new function w/ our custom logic.